### PR TITLE
CON-1193 - conclave init - Added a logging framework to be used as default by the projects generated using the conclave-init tool

### PR DESCRIPTION
**Pull request by filipesoliveira**
The projects generated by `conclave-init` will now have a default logging framework. 
Updated io.spring.dependency-management to the latest version (1.1.0).
Added a check to ensure the host generates logs by default.

### CODE REVIEW

**Comment by filipesoliveira** [comment]
_Can you look into updating test-conclave-init.sh to capture the output of the boot jar run to make sure it logged the missing bits._
Sorry, @shamsasari. I did not understand what you mean by "capture the output of the boot jar run to make sure it logged the missing bits."

**Comment by shamsasari** [comment]
I mean, can we test that the logging is working? We test conclave init in the test-conclave-init.sh script. Is it possible to update it so that when the generated code is executed, we assert that the host process logged the lines which previously was not being logged. This should be possible by simply redirecting to grep (via a file or direct via a pipe).

File: test-conclave-init.sh [hunk 1]
```diff
@@ -39,7 +39,7 @@ sed -i "s/repositories {/repositories {\nmaven { url = '..\/repo' }/" settings.g
 echo Run Java project host and client
 
 # NOTE: this command is only run in mock mode, so that the signer can be provided below.
-./gradlew :host:run & _PID=$!
+./gradlew :host:bootRun & _PID=$!
```
Comment (shamsasari) [inline test-conclave-init.sh:42]: Do we need to update the docs?

Comment (filipesoliveira) [reply test-conclave-init.sh:42]: Docs have been updated.

**Comment by shamsasari** [comment]
Test failing for some reason.

**Comment by filipesoliveira** [comment]
@shamsasari, all tests are now green.

**Review by shamsasari — APPROVED** [review]
LGTM

**Pull request merged**